corehq/apps/locations/static/locations/js/location_tree.async.js

```diff
@@ -146,4 +166,140 @@ function LocationModel(data, root, depth) {
         hqImport('analytix/js/google').track.event('Organization Structure', '+ New _______');
         return true;
     };
+
+    this.remove_elements_after_action = function(button) {
+        $(button).closest('.loc_section').remove();
+    };
+
+    this.archive_success_message = _.template(django.gettext("You have successfully archived the location <%=name%>"));
+
+    this.delete_success_message = _.template(django.gettext(
+        "You have successfully deleted the location <%=name%> and all of its child locations"
+        )
+    );
+
+    this.delete_error_message = django.gettext("An error occurred while deleting your location. If the problem persists, please report an issue");
+
+    this.loc_archive_url = function(loc_id) {
+        var initial_page_data = hqImport('hqwebapp/js/initial_page_data');
+        var template = initial_page_data.reverse('archive_location');
+        return template.replace('-locid-', loc_id);
+    };
+
+    this.loc_unarchive_url = function(loc_id) {
+        var initial_page_data = hqImport('hqwebapp/js/initial_page_data');
+        var template = initial_page_data.reverse('unarchive_location');
+        return template.replace('-locid-', loc_id);
+    };
+
+    this.loc_delete_url = function(loc_id) {
+        var initial_page_data = hqImport('hqwebapp/js/initial_page_data');
+        var template = initial_page_data.reverse('delete_location');
+        return template.replace('-locid-', loc_id);
+    };
+
+    this.loc_descendant_url = function(loc_id) {
+        var initial_page_data = hqImport('hqwebapp/js/initial_page_data');
+        var template = initial_page_data.reverse('location_descendants_count');
+        return template.replace('-locid-', loc_id);
+    };
+
+    this.archive_loc = function(button, name, loc_id) {
+        var archive_location_modal = $('#archive-location-modal')[0];
+
+        function archive_fn() {
+            $(button).disableButton();
+            $.ajax({
+                type: 'POST',
+                url: loc.loc_archive_url(loc_id),
+                dataType: 'json',
+                error: 'error',
+                success: function () {
+                    alert_user(loc.archive_success_message({"name": name}), "success");
+                    loc.remove_elements_after_action(button);
+                    if (hqImport('hqwebapp/js/toggles').toggleEnabled('LOCATION_SEARCH')) {
+                        reloadSelect(); // jshint ignore:line
```

Comment:
Looks like I gave you the wrong directive to add here. According to https://stackoverflow.com/questions/27732209/turning-off-eslint-rule-for-a-specific-line#29592334, look like the comment we want to add is `// eslint-disable-line no-undef`. (This is because stickler uses a linter called eslint and not jshint as I'd thought.)